compaction/compaction_group_view.hh

```diff
@@ -38,8 +38,8 @@ public:
     // min threshold as defined by table.
     virtual unsigned min_compaction_threshold() const noexcept = 0;
     virtual bool compaction_enforce_min_threshold() const noexcept = 0;
-    virtual const sstables::sstable_set& main_sstable_set() const = 0;
-    virtual const sstables::sstable_set& maintenance_sstable_set() const = 0;
+    virtual future<lw_shared_ptr<const sstables::sstable_set>> main_sstable_set() const = 0;
+    virtual future<lw_shared_ptr<const sstables::sstable_set>> maintenance_sstable_set() const = 0;
```

Comment:
Do these futurizations (this and the previous one) create opening for compaction races? E.g. one compaction fiber is retrieving candidates, another one is started and manages to start. Do we have locks to prevent such scenarios?